services/geocoding.py

```diff
@@ -1,41 +1,55 @@
 import re
-
+import json
+import googlemaps
 from dataclasses import dataclass
-from typing import List, Mapping
-
+from typing import Callable, List, Mapping
 from returns.io import impure_safe
-
 from config.settings import Settings
 from models.point_of_interest import PointOfInterest
 
 
+
+
 @dataclass
 class Point:
     lat: str
     lng: str
 
+def make_geocode_request(address: str, gmaps):
+   return gmaps.geocode(address)
 
+def init_google_maps(key):
+    return googlemaps.Client(key=key)
 @dataclass
 class GeoCodingProcessor:
 
     # Injected dependencies
     settings: Settings
-
+    make_geocode_request: Callable = make_geocode_request
+    init_google_maps: Callable = init_google_maps
+    
     @impure_safe
     def enhance(self, entries: List[PointOfInterest]) -> List[PointOfInterest]:
         """
         For entries with missing lat/lng but present address
         we can find lat/lng using a geocoding API.
         """
+
+        #Create client instance of gmaps
+        gmaps = self.init_google_maps(self.settings.developer_key)
+
```

Comment:
Btw, when you're geocoding, it might be a good idea to actually include the city and the country when available because it might not be a part of the address. For example, if address is `Baker street, 10` and `point.city = 'Warsaw` and `point.country = 'Poland'` then the address you're geocoding should be `Baker street, 10, Warsaw, Poland` . And of course, you should append those only if they are present and if they are not duplicated (already in the address) so we don't end up with `Baker street, 10, Warsaw, Poland, Warsaw, Poland`

Reply:
this would be another enhancement beyond the scope of this ticket.